ddtrace/ext/ci_visibility/api.py

```diff
@@ -0,0 +1,435 @@
+            item_id,
+            override_status,
+            force_finish_children,
+        )
+        core.dispatch(
+            "ci_visibility.module.finish", (CIModule.FinishArgs(item_id, override_status, force_finish_children),)
+        )
+
+
+class CISuite(_CIVisibilityAPIBase):
+    class DiscoverArgs(NamedTuple):
+        suite_id: CISuiteId
+        codeowners: Optional[List[str]] = None
+        source_file_info: Optional[CISourceFileInfo] = None
+
+    class FinishArgs(NamedTuple):
+        suite_id: CISuiteId
+        force_finish_children: bool = False
+        override_status: Optional[CITestStatus] = None
+        is_itr_skipped: bool = False
+
+    @staticmethod
+    @_catch_and_log_exceptions
+    def discover(
+        item_id: CISuiteId,
+        codeowners: Optional[List[str]] = None,
+        source_file_info: Optional[CISourceFileInfo] = None,
+    ):
+        """Registers a test suite with the CI Visibility service."""
+        log.debug("Registering suite %s, source: %s", item_id, source_file_info)
+        core.dispatch("ci_visibility.suite.discover", (CISuite.DiscoverArgs(item_id, codeowners, source_file_info),))
+
+    @staticmethod
+    @_catch_and_log_exceptions
+    def start(item_id: CISuiteId):
+        log.debug("Starting suite %s", item_id)
+        core.dispatch("ci_visibility.suite.start", (item_id,))
+
+    @staticmethod
+    @_catch_and_log_exceptions
+    def finish(
+        item_id: CISuiteId,
+        force_finish_children: bool = False,
+        override_status: Optional[CITestStatus] = None,
+        is_itr_skipped: bool = False,
+    ):
+        log.debug(
+            "Finishing suite %s, override_status: %s, force_finish_children: %s",
+            item_id,
+            force_finish_children,
+            override_status,
+        )
+        core.dispatch(
+            "ci_visibility.suite.finish",
+            (CISuite.FinishArgs(item_id, force_finish_children, override_status, is_itr_skipped),),
+        )
+
+    @staticmethod
+    @_catch_and_log_exceptions
+    def mark_itr_skipped(item_id: CITestId, force_finish_children: bool = False):
+        log.debug("Marking test %s as skipped by ITR", item_id)
+        CISuite.finish(item_id, force_finish_children, CITestStatus.SKIP, is_itr_skipped=True)
+
+    @staticmethod
+    @_catch_and_log_exceptions
+    def add_coverage_data(item_id: CITestId, coverage_data: Dict[str, Any]):
+        log.debug("Adding coverage data for suite %s: %s", item_id, coverage_data)
+
+
+class CITest(_CIVisibilityAPIBase):
+    class DiscoverArgs(NamedTuple):
+        test_id: CITestId
+        codeowners: Optional[List[str]] = None
+        source_file_info: Optional[CISourceFileInfo] = None
+
+    @staticmethod
+    @_catch_and_log_exceptions
+    def discover(
+        item_id: CITestId,
+        codeowners: Optional[List[str]] = None,
+        source_file_info: Optional[CISourceFileInfo] = None,
+        is_early_flake_detection: bool = False,
```

Comment:
interesting point here: do API consumers decide if early flake detection is enabled or should the library itself decide it?